infrastructure-playbooks/make-osd-partitions.yml

```diff
@@ -0,0 +1,52 @@
+---
+# This playbook will make custom partition layout for your osd hosts.
+# You should define `devices` variable for every host.
+#
+# For example, in host_vars/hostname1
+# ```
+# devices:
+# - device_name: sdb
+#   partitions:
+#   - index: 1
+#     size: 10G
+#     type: data
+#   - index: 2
+#     size: 5G
+#     type: journal
+# - device_name: sdc
+#   partitions:
+#   - index: 1
+#     size: 10G
+#     type: data
+#   - index: 2
+#     size: 5G
+#     type: journal
+# ```
+- hosts: all
+  vars:
+    journal_typecode: 45b0969e-9b03-4f30-b4c6-b4b80ceff106
+    data_typecode: 4fbd7e29-9d25-41b8-afd0-062c0ceff05d
+
+
+  tasks:
+  - name: install sgdisk(gdisk)
+    package:
+      name: gdisk
+      state: present
+    register: pkg_result
+
+  - name: erase all previous partitions(dangerous!!!)
+    command: /usr/sbin/sgdisk --zap-all -- /dev/{{item.device_name}}
+    with_items: "{{ devices }}"
+    register: erase_result
+    when: pkg_result|succeeded
```

Comment:
Why don't you fail instead?
If sgdisk is not present we should fail and tell users to install it.